src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker.cs

```diff
@@ -704,8 +704,13 @@ public CompilationInfo(Compilation compilation, bool hasSuccessfullyLoaded, Comp
             {
                 try
                 {
-                    // if HasAllInformation is false, then this project is always not completed.
-                    var hasSuccessfullyLoaded = this.ProjectState.HasAllInformation;
+                    // Project is complete only if the following are all true:
+                    //  1. HasAllInformation flag is set for the project
```

Comment:
I think there are some tests in SolutionTests.cs that test this flag, so you could add a test there.

Reply:
Done, added a test. Thanks!